exonum-light-client/src/main/java/com/exonum/client/ExonumHttpClient.java

```diff
@@ -106,6 +115,68 @@ public String getUserAgentInfo() {
     });
   }
 
+  @Override
+  public BlockResponse getBlockByHeight(long height) {
+    HttpUrl url = urlBuilder()
+        .encodedPath(BLOCK)
+        .addQueryParameter("height", String.valueOf(height))
+        .build();
+    Request request = getRequest(url);
+
+    return blockingExecuteAndParse(request, ExplorerApiHelper::parseGetBlockResponse);
+  }
+
+  @Override
+  public BlocksResponse getBlocks(int count, boolean skipEmpty, Long heightMax,
+      boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, heightMax, withBlocksTime);
+  }
+
+  @Override
+  public BlocksResponse getLastBlocks(int count, boolean skipEmpty, boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, null, withBlocksTime);
+  }
+
+  @Override
+  public Block getLastBlock() {
+    BlocksResponse response = doGetBlocks(1, false, null, false);
+
+    return response.getBlocks()
+        .stream()
+        .findFirst()
+        .orElseThrow(() -> new AssertionError("Should never happen"));
+  }
+
+  @Override
+  public Optional<Block> getLastNotEmptyBlock() {
+    BlocksResponse response = doGetBlocks(1, true, null, false);
+
+    return response.getBlocks()
+        .stream()
+        .findFirst();
+  }
+
+  private BlocksResponse doGetBlocks(int count, boolean skipEmpty, Long heightMax,
+      boolean withBlocksTime) {
+    checkArgument(count <= MAX_BLOCKS_PER_REQUEST,
```

Comment:
Check it is strictly positive?

Reply:
⬆️ It can be 0 if we would like to see the height only

Reply:
Not sure I understand. What will 0 get us?

Reply:
If `count ==0` then we would like to see the height only. It's the only way to get current blockchain height 🤷‍♀️